File: backend/src/api/health.py
# ==========================================================
# src/api/v1/endpoints/health.py
# ==========================================================

from fastapi import APIRouter

from src.core.config import settings


router = APIRouter(
    prefix="/health",
    tags=["Health"],
)


@router.get("")
def health_check() -> dict:
    """
    Application health and metadata endpoint.

    Used by:
    - Docker healthcheck
    - Nginx/container monitoring
    - Load balancers
    - Deployment verification
    - Frontend runtime metadata resolution
    """

    return {
        "status": "healthy",
        "service": "Finora API",
        "app_name": settings.APP_NAME,
        "version": settings.APP_VERSION,
    }

File: frontend/src/api/health.ts
// ==========================================================
// src/api/health.ts
// ==========================================================
//
// Finora Backend Health API
//
// Purpose:
//   Resolve the actual running backend application metadata.
//
// Version source of truth:
//
//   Git Release Tag
//        ↓
//   APP_VERSION
//        ↓
//   Backend Docker Image
//        ↓
//   /api/v1/health
//        ↓
//   Frontend
//
// The frontend does NOT use VITE_APP_VERSION as the
// runtime application version.
//
// ==========================================================

import { publicApi } from "./publicClient";

// ==========================================================
// Health Status
// ==========================================================

export type HealthStatus = "healthy" | "unhealthy";

// ==========================================================
// Health Response
// ==========================================================

export interface HealthResponse {
  status: HealthStatus;
  service: string;
  app_name: string;
  version: string;
}

// ==========================================================
// Get Backend Health
// ==========================================================
//
// Endpoint:
//
//   GET /api/v1/health
//
// Because publicApi already uses:
//
//   /api/v1
//
// this function only needs:
//
//   /health
//
// ==========================================================

export const getHealth = async (): Promise<HealthResponse> => {
  const response = await publicApi.get<HealthResponse>("/health");

  return response.data;
};

